lib/index.js

```diff
@@ -9,20 +9,27 @@ const escapeStringRegexp = require('escape-string-regexp');
 const BUILD_DIR_REGEXP = new RegExp('(' + escapeStringRegexp(path.sep) + ')?build(' + escapeStringRegexp(path.sep) + ')?$');
 const IGNORED_FILE_MESSAGE_REGEXP = /(?:File ignored by default\.)|(?:File ignored because of a matching ignore pattern\.)/;
 
-
 /**
  * Calculates the severity of a eslint.linter.verify result
- * @param {Array} result Eslint verify result array
- * @returns {Number} If the returned number is greater than 0 the result contains errors.
+ * @param {Array} result ESLint's verify() result array
+ *    @see: http://eslint.org/docs/developer-guide/nodejs-api#linter
+ *
+ * @returns {Number} accumulatedSeverity The total severity from of the list of results
+ *    0 indicates all-clear
+ *    1 indicates a warning-level result
+ *    > 1 indicates an error-level result
  */
 function getResultSeverity(resultMessages = []) {
+  let severity;
+
   return resultMessages.reduce((accumulatedSeverity, message) => {
+    severity = message.severity || 0;
```

Comment:
why did you move the `let severity` out of the lambda? this often indicates that the function has side-effects but I don't see any in this case.